src/RepositoryBase.cs

```diff
@@ -128,6 +128,12 @@ public Task DeleteAsync(T entity, CancellationToken cancellationToken = default)
             return Db.SaveChangesAsync();
         }
 
+        public Task DeleteWhereAsync(Expression<Func<T, bool>> predicate, CancellationToken cancellationToken = default)
+        {
+            Entities.RemoveRange(Entities.Where(predicate));
+            return Db.SaveChangesAsync();
```

Comment:
Don't forget to support the cancellation token!
```suggestion
            return Db.SaveChangesAsync(cancellationToken);
```

In fact, while you're here can you make sure the others also support cancellation?

Reply:
On it